hooks/hookexecution/enricher.go

```diff
@@ -0,0 +1,184 @@
+package hookexecution
+
+import (
+	"encoding/json"
+
+	"github.com/buger/jsonparser"
+	"github.com/prebid/openrtb/v17/openrtb2"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/hooks/hookanalytics"
+	jsonpatch "gopkg.in/evanphx/json-patch.v4"
+)
+
+const (
+	// traceLevelBasic excludes debugmessages and analytic_tags from output
+	traceLevelBasic trace = "basic"
+	// traceLevelVerbose sets maximum level of output information
+	traceLevelVerbose trace = "verbose"
+)
+
+// Trace controls the level of detail in the output information returned from executing hooks.
+type trace string
+
+func (t trace) isBasicOrHigher() bool {
+	return t == traceLevelBasic || t.isVerbose()
+}
+
+func (t trace) isVerbose() bool {
+	return t == traceLevelVerbose
+}
+
+// EnrichExtBidResponse adds debug and trace information returned from executing hooks to the ext argument.
+// In response the outcome is visible under the key response.ext.prebid.modules.
+//
+// Debug information is added only if the debug mode is enabled by request and allowed by account (if provided).
+// The details of the trace output depends on the value in the bidRequest.ext.prebid.trace field.
+func EnrichExtBidResponse(
+	ext json.RawMessage,
+	stageOutcomes []StageOutcome,
+	bidRequest *openrtb2.BidRequest,
+	account *config.Account,
+) (json.RawMessage, error) {
+	modules, err := GetModulesJSON(stageOutcomes, bidRequest, account)
+	if err != nil {
+		return ext, err
+	} else if modules == nil {
+		return ext, nil
+	}
+
+	response := json.RawMessage(`{"prebid":{"modules":` + string(modules) + `}}`)
```

Comment:
Please use json.Marshal here too to ensure the modules do not contain an attack vector.